packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,332 @@
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import { Construct, IResource, Lazy, Resource, Stack, Token, Duration } from '@aws-cdk/core';
+import { CfnDistribution } from './cloudfront.generated';
+import { Origin } from './origin';
+import { CacheBehavior } from './private/cache-behavior';
+
 /**
  * Interface for CloudFront distributions
  */
-export interface IDistribution {
+export interface IDistribution extends IResource {
+  /**
+   * The domain name of the Distribution, such as d111111abcdef8.cloudfront.net.
+   *
+   * @attribute
+   */
+  readonly domainName: string;
```

Comment:
If this is a CFN attribute it should start with `distribution`, so `distributionDomainName`

Reply:
1. Out of curiosity, why? This goes counter to good advise/patterns elsewhere of removing redundant naming prefixing.
2. More importantly, this isn't new code (despite what the diff is showing). This interface and the names are part of the existing `CloudFrontWebDistribution` hierarchy and are referenced both in other L2s as well as customer code. Renaming this will be a breaking change. 

Do you believe renaming this is worth a breaking change?

Reply:
> Out of curiosity, why? This goes counter to good advise/patterns elsewhere of removing redundant naming prefixing.

The rationale is that we need a way to differentiate between attributes and properties and we standardized around this convention.